src/main/java/com/google/firebase/database/snapshot/BigIntegerNode.java

```diff
@@ -0,0 +1,60 @@
+package com.google.firebase.database.snapshot;
```

Comment:
Lot of the comments made on `BigDecimalNode` are also applicable here. Please revise as appropriate.

Reply:
I'd add the tests, but I don't see the idea behind how you test your code. The structure is not clear to find an easy entry-point.

Reply:
`NodeTest` might be a good place to start. Also check `MapperTest`. See if you can serialize/deserialize some made up big integers and big decimals via the new code.